### PR TITLE
added a command to edit deadline of tasks

### LINKED ISSUE #16: As a HR Manager, I can change the deadline for an existing task

### PR DESCRIPTION
**Pull request by laybach**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditDeadlineCommand.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     * Change the deadline of the task of a user.
+     */
+    public EditDeadlineCommand(Index index, Deadline deadline) {
+        requireAllNonNull(index, deadline);
+        this.index = index;
+        this.deadline = deadline;
+    }
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) throws CommandException {
```
Comment (ziyi22) [inline src/main/java/seedu/address/logic/commands/EditDeadlineCommand.java:52]: This method is not coverage by tests.

File: src/main/java/seedu/address/logic/commands/EditDeadlineCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        Person taskOwner = personList.get(index.getZeroBased());
+
+        if (!taskOwner.hasTask()) {
+            throw new CommandException(MESSAGE_TASK_DOES_NOT_EXIST);
+        }
+
+        Task task = taskOwner.getTask();
+        task.editDeadline(deadline);
+        model.commitAddressBook();
+        return new CommandResult(String.format(MESSAGE_MARK_TASK_SUCCESS,
```
Comment (ziyi22) [inline src/main/java/seedu/address/logic/commands/EditDeadlineCommand.java:69]: Can improve test coverage.

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -97,6 +97,14 @@ public boolean hasTask(String taskName) {
         }
     }
 
+    /**
+     * Check if person has specified task assigned
+     * @return boolean
+     */
+    public boolean hasTask() {
```
Comment (howenyap) [inline src/main/java/seedu/address/model/person/Person.java]: The isBusy() method is already available on line 83.

File: src/test/java/seedu/address/logic/parser/EditDeadlineCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package seedu.address.logic.parser;
+
+//import static org.junit.jupiter.api.Assertions.assertEquals;
+//import static org.junit.jupiter.api.Assertions.assertFalse;
+//import static org.junit.jupiter.api.Assertions.assertNotEquals;
+//import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (howenyap) [inline src/test/java/seedu/address/logic/parser/EditDeadlineCommandParserTest.java:6]: Remove code instead of commenting them out.

**Review by howenyap — COMMENTED** [review]
Proposed some changes.

File: src/main/java/seedu/address/model/person/Person.java (hunk 3 above)
Comment (ziyi22) [inline src/main/java/seedu/address/model/person/Person.java]: Can use Person.isBusy() instead.

File: src/test/java/seedu/address/logic/commands/EditDeadlineCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+//import static org.junit.jupiter.api.Assertions.assertFalse;
+//import static org.junit.jupiter.api.Assertions.assertNotEquals;
+//import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+class EditDeadlineCommandTest {
+
+    @Test
+    void execute() {
```
Comment (KiatLun) [inline src/test/java/seedu/address/logic/commands/EditDeadlineCommandTest.java:13]: Remember to implement tests for the execute method

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 6]
```diff
@@ -59,6 +59,11 @@ public void parseCommand_assignTask() throws Exception {
         assertEquals(new AssignTaskCommand(task, INDEX_FIRST_PERSON), command);
     }
 
+    @Test
+    public void parseCommand_editDeadline() throws Exception {
```
Comment (ziyi22) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:63]: This method has not implemented yet.

**Review by howenyap — APPROVED** [review]
LGTM

**Pull request merged**